docs/general/ad-filtering/create-own-filters.md

```diff
@@ -1737,6 +1740,210 @@ As a response to blocked request AdGuard returns a short video placeholder.
+
+The base priority weight of any rule is 1. If the priority is a floating point number, it will be **rounded up** to the smaller integer greater than or equal to the calculated weight.
+
+> **Compatibility with different versions of AdGuard**
+>
+> Numerical priorities for rules are available in `tsurlfilter` since version `v2.1.0`.
+
+#### 1. Basic modifiers, the presence of each adds 1 to the priority: {#priority-category-1}
+
+[//]: # (Please keep them sorted)
+
+ * [`$app`](#app-modifier) with forbidden applications using `~`,
+ * [`$denyallow`](#denyallow-modifier),
+ * [`$dnsrewrite`](#dnsrewrite-modifier),
+ * [`$domain`](#domain-modifier) with forbidden domains using `~`,
+ * `$first-party`,
+ * [`$header`](#header-modifier),
+ * [`$match-case`](#match-case-modifier),
+ * [`$method`](#method-modifier) with forbidden methods using `~`,
+ * [`$third-party`](#third-party-modifier),
+ * [`$to`](#to-modifier),
+ * restricted [content-types](#content-type-modifiers) with `~`.
+
+For forbidden domain, app or content-type, we add 1 for the presence of the modifier itself disregarding the number of forbidden domains or content-types, because the scope of the rule is infinitely large anyway. In other words, by banning several domains and content-types, we only *insignificantly* narrowed the scope of the rule.
+
+#### 2. Defined content-type modifiers, defined methods, $popup, special exceptions: {#priority-category-2}
+
+All allowed content types:
+[//]: # (Please keep them sorted)
+
+* [`$document`](#document-modifier),
+* [`$font`](#font-modifier),
+* [`$image`](#image-modifier),
+* [`$media`](#media-modifier),
+* [`$object`](#object-modifier),
+* [`$other`](#other-modifier),
+* [`$ping`](#ping-modifier),
+* [`$script`](#script-modifier),
+* [`$stylesheet`](#stylesheet-modifier),
+* [`$subdocument`](#subdocument-modifier),
+* [`$websocket`](#websocket-modifier),
+* [`$xmlhttprequest`](#xmlhttprequest-modifier);
+
+This also includes rules that implicitly add the modifier `$document`:
+* [`$popup`](#popup-modifier);
+
+or special exceptions that implicitly add `$document,subdocument`:
+* [`$content`](#content-modifier),
+* [`$elemhide`](#elemhide-modifier),
+* [`$extension`](#extension-modifier),
+* [`$jsinject`](#jsinject-modifier),
+* [`$specifichide`](#specifichide-modifier),
+* [`$urlblock`](#urlblock-modifier),
+* [`$genericblock`](#genericblock-modifier),
+* [`$generichide`](#generichide-modifier);
```

Comment:
`generichide` and `genericblock` are also present in `Specific exceptions`

Reply:
@sxgunchenko you mean i need delete sub-item "[Generic rules](#exception-modifiers-generic-rules)" ?

Reply:
Maybe, I didn't get it right. Here's said that those modifiers just imply the `document` content type for a rule, right?